03-bash/task1.sh

```diff
@@ -0,0 +1,157 @@
+        "MSG3") echo "Забагато аргументiв" >&2;;
+        "MSG4") echo "Не вдалося створити папку виводу" >&2;;
+        "MSG5") echo "Не вдалося створити файл" >&2;;
+    esac
+    else
+    case "$1" in 
+        "MSG1") echo "-n parameter must be equal or greater than 1" >&2;;
+        "MSG2") echo "-n parameter must be an integer value" >&2;;
+        "MSG3") echo "Too many arguments" >&2;;
+        "MSG4") echo "Failed to create output folder" >&2;;
+        "MSG5") echo "Failed to create output file" >&2;;
+    esac;
+    fi
+}
+
+function Locale()
+{
+    if [ "$LANG" = "uk_UA.UTF-8" ]
+    then
+        LOCALE=UA
+    fi
+}
+
+function checkN() 
+{
+    if [[ "$1" =~ ^[0-9]+$ ]]
+    then
+        if [ "$1" -ge 1 ]
+        then
+            PARAM_N=let "$1-1"
+        else
+            printLocalizedMessages "MSG1"
+            exit 1
+        fi
+    else
+        printLocalizedMessages "MSG2"
+        exit 1
+    fi
+}
+
+function args()
+{
+    if [ "$#" -gt 3 ]
+    then
+        printLocalizedMessages "MSG3"
+        exit 1
+    fi
+    while [ -n "$1" ]
+    do
+        case "$1" in
+            -h|--help)  usage
+                        exit 0;;
+            -n)         PARAM_N=checkN $2
+                        shift;;
+            *)          FILE_PATH=$1;;
+        esac
+        shift
+    done
+}
+
+function getHWINFO()
+{
+    local result
+
+    set -o pipefail
+
+    echo -e "---- Hardware ----"  > ${FILE_PATH}
+    echo -e "CPU: \"$(dmidecode -s processor-version || echo "Unknown")\"" >> ${FILE_PATH}
+    echo -e $(dmidecode -t 17 | awk '( /Size/ && $2 ~ /^[0-9]+$/ ) { x+=$2 } END{ print "RAM: " x "MB" }' || echo "RAM: Unknown") >> ${FILE_PATH}
+    echo -e "Motherboard: \"$(dmidecode -s baseboard-manufacturer || echo "Unknown")\", \"$(dmidecode -s baseboard-product-name || echo "Unknown")\"" >> ${FILE_PATH}
+    echo -e "System Serial Number: $(dmidecode -s system-serial-number || echo "Unknown")" >> ${FILE_PATH}
+    echo -e "---- System ----" >> ${FILE_PATH}
+    echo -e "OS Distribution: $(cat /etc/*-release | grep -i description | sed 's\DISTRIB_DESCRIPTION=\\g' || echo "Unknown")" >> ${FILE_PATH}
+    echo -e "Kernel version: $(uname -r || echo "Unknown")" >> ${FILE_PATH}
+    echo -e "Installation date: $(tune2fs -l $(df / | tail -1 | cut -f1 -d' ') | grep -i "filesystem created" | sed 's/Filesystem created:       //g' || echo "Unknown")" >> ${FILE_PATH}
+    echo -e "Hostname: $(hostname || echo "Unknown")" >> ${FILE_PATH}
+    echo -e "Uptime: $(uptime -p || echo "Unknown")" >> ${FILE_PATH}
+    echo -e "Processes running: $(ps aux --no-heading | wc -l || echo "Unknown")" >> ${FILE_PATH}
+    echo -e "User logged in: $(who | wc -l || echo "Unknown")" >> ${FILE_PATH}
+    echo -e "---- Network ----" >> ${FILE_PATH}
+    for item in $(ifconfig | cut -d" " -f1 | awk NF | cut -d":" -f1)
+    do
+	    result=$(ip -f inet address show $item | grep inet | awk '{ print $2 }')
+        if [ ! -z $result ]
```

Comment:
not condition in if - else -fi look strange.
You can do it without using '!'
Just by changing the code blocks.